cobalt/shell/common/shell_paths.cc

```diff
@@ -28,16 +28,29 @@
 #include "base/nix/xdg_util.h"
 #endif
 
+#if BUILDFLAG(IS_COBALT_HERMETIC_BUILD)
+#include "starboard/configuration_constants.h"
+#include "starboard/system.h"
+#endif
+
 namespace content {
 
 namespace {
 
 bool GetDefaultUserDataDirectory(base::FilePath* result) {
-#if BUILDFLAG(IS_LINUX)
-  std::unique_ptr<base::Environment> env(base::Environment::Create());
-  base::FilePath config_dir(base::nix::GetXDGDirectory(
-      env.get(), base::nix::kXdgConfigHomeEnvVar, base::nix::kDotConfigDir));
-  *result = config_dir.Append("content_shell");
+#if BUILDFLAG(IS_COBALT_HERMETIC_BUILD)
+  std::vector<char> path(kSbFileMaxPath, 0);
+  bool success =
+      SbSystemGetPath(kSbSystemPathFilesDirectory, path.data(), path.size());
+  DCHECK(success);
+  if (!success) {
+    DLOG(ERROR) << "kSbSystemPathFilesDirectory not defined.";
+    return false;
```

Comment:
What happens when this returns false? Can the app reasonably continue? If not, the DCHECK above should be a CHECK